File: src/main/resources/web/js/h5chart/H5Chart.js
/*******************************************************************************
 * File: H5Chart.js 
 * Copyright (C) 2009-2011 
 * Author: L.D.Nigro 
 * Contributor(s):
 * 
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * 
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 * 
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 ******************************************************************************/

zk.$package('H5Chart');

h5chart.H5Chart = zk.$extends(zk.Widget, {
	
	_rm: null,
	
	_regionManager: 'false',
	
	isRegionManager : function() {
		return this._regionManager;
	},
	
	setRegionManager : function(value) {
		this._regionManager = value;
		this._rm=new RegionManager();
		this._rm.init();
	},
	
	bind_: function (desktop, skipper, after) {
		this.$supers('bind_', arguments);
		
		var s=this.uuid;

		var canvas=document.getElementById(s);
		canvas.regions=this._rm.regions;
		canvas.regionManager=this._rm;
		canvas.onclick=onClick;
		canvas.onmousemove=onMouseMove;
		canvas.onmousedown=onMouseDown;
		canvas.onmouseup=onMouseUp;
	},
	
	detach: function() {
		if(this._rm){
			rm.clear();
		}
		this._rm= null;
		this.clear();
	} 

});

File: src/main/java/h5chart/H5Chart.java
/*******************************************************************************
 * File: H5Chart.java 
 * Copyright (C) 2009-2011 
 * Author: L.D.Nigro 
 * Contributor(s):
 * 
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * 
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 * 
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 ******************************************************************************/

package h5chart;

import org.zkoss.zk.ui.HtmlBasedComponent;

public class H5Chart extends HtmlBasedComponent {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	private String _regionManager="false";
	
	public boolean isRegionManager() {
		return new Boolean(_regionManager);
	}

	public void setRegionManager(boolean regionManager) {
		String an=Boolean.toString(regionManager);
		if(!_regionManager.equals(an)){
			_regionManager = an;
			smartUpdate("regionManager", _regionManager);
		}
	}
	
	protected void renderProperties(org.zkoss.zk.ui.sys.ContentRenderer renderer)
			throws java.io.IOException {
		super.renderProperties(renderer);
		
		render(renderer, "regionManager", true);
	}
	
}
